backend/src/middleware/validation/validationMiddleware.js

```diff
@@ -59,13 +59,23 @@ const validateUpdatePost = async (resolve, root, args, context, info) => {
 
 const validateReport = async (resolve, root, args, context, info) => {
   const { resourceId } = args
-  const { user, driver } = context
+  const { user } = context
   if (resourceId === user.id) throw new Error('You cannot report yourself!')
+  return resolve(root, args, context, info)
+}
+
+const validateReview = async (resolve, root, args, context, info) => {
+  const { resourceId } = args
+  const { user, driver } = context
+  if (resourceId === user.id) throw new Error('You cannot review yourself!')
   const session = driver.session()
   const reportQueryRes = await session.run(
     `
-      MATCH (:User {id:$submitterId})-[:REPORTED]->(resource {id:$resourceId}) 
-      RETURN labels(resource)[0] as label
+      MATCH (resource {id: $resourceId})
+      WHERE resource:User OR resource:Post OR resource:Comment
+      OPTIONAL MATCH (:User)-[filed:FILED]->(:Report {closed: false})-[:BELONGS_TO]->(resource)
```

Comment:
Are there ever situations in which ~we have a `:Report` without a user who filed it or~ no `author:User` who `:WROTE` the resource?

Reply:
I don't think so... I think this `OPTIONAL MATCH` is because the resource is not always a `Post` or `Comment` that was written by someone. If it is then we are checking that a moderator cannot review there own `Post|Comment`, see lines 100-105